HotelBooking/HotelBooking.App/Operations/OperationBase.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using HotelBooking.Domain.Models;
+using HotelBooking.App.DataObjects;
+using HotelBooking.App.Enums;
+
+namespace HotelBooking.App.Operations
+{
+    public abstract class OperationBase : IOperation
+    {
+        public abstract bool IsRequiredToSucceed { get; }
+        public abstract string OperationName { get; }
+        public abstract Func<Reservation, bool> Execute { get; }
+
+        public OperationResult OperationResult { get; set; }
+
+        public void Process(Reservation reservation, BookingResult bookingResult)
+        {
+            var executionResult = Execute(reservation) ? Result.Success : Result.Failure;
+
+            OperationResult = new OperationResult
+            {
+                ExecutionResult = executionResult,
+                OperationName = OperationName,
+                ShouldAbortBookingProcess = IsRequiredToSucceed && executionResult == Result.Failure
+            };           
+            bookingResult.OperationResults.Add(OperationResult);

```

Comment:
Mysle ze pojedyncza operacja nie powinna wiedziec jak agregowac bledy. Powinna to robic klasa wyzej - OperationManager. chetnie to rozwine jak nie jest to jasne ale tak na szybko logika agregacji moze sie zmieniac z czasem co nie powinno miec wplywu na operacje. pozatym pozbedziesz modyfikowania bookingResult w tej metodzie, ktory dobrze by bylo jak by byl immutable.